src/services/completions.ts

```diff
@@ -826,6 +830,174 @@ function jsdocCompletionInfo(entries: CompletionEntry[]): CompletionInfo {
+                insertText = insertText.slice(1);
+                snippetText = snippetText.slice(1);
+            }
+            return {
+                name: insertText,
+                kind: ScriptElementKind.parameterElement,
+                sortText: SortText.LocationPriority,
+                insertText: isSnippet ? snippetText : undefined,
+                isSnippet,
+            };
+        }
+        else if (param.parent.parameters.indexOf(param) === paramTagCount) { // Destructuring parameter; do it positionally
+            const paramPath = `param${paramTagCount}`;
+            const insertTextResult =
+                generateJSDocParamTagsForDestructuring(paramPath, param.name, param.initializer, isJs, /*isSnippet*/ false);
+            const snippetTextResult =
+                generateJSDocParamTagsForDestructuring(paramPath, param.name, param.initializer, isJs, /*isSnippet*/ true);
+            for (let i = 1; i < insertTextResult.length; i++) {
+                insertTextResult[i] = `* ${insertTextResult[i]}`;
+                snippetTextResult[i] = `* ${snippetTextResult[i]}`;
+            }
+            let insertText = insertTextResult.join(getNewLineCharacter(options));
+            let snippetText = snippetTextResult.join(getNewLineCharacter(options));
+            if (tagNameOnly) { // Remove `@`
+                insertText = insertText.slice(1);
+                snippetText = snippetText.slice(1);
+            }
+            return {
+                name: insertText,
+                kind: ScriptElementKind.parameterElement,
+                sortText: SortText.LocationPriority,
+                insertText: isSnippet ? snippetText : undefined,
+                isSnippet,
+            };
+        }
+    });
+}
+
+function generateJSDocParamTagsForDestructuring(
+    path: string,
+    pattern: BindingPattern,
+    initializer: Expression | undefined,
+    isJs: boolean,
+    isSnippet: boolean): string[] {
+    if (!isJs) {
+        return [getJSDocParamAnnotation(path, initializer, isJs, /*isObject*/ false, isSnippet, { tabstop: 1 })];
+    }
+    return patternWorker(path, pattern, { tabstop: 1 });
+
+    function patternWorker(path: string, pattern: BindingPattern, counter: TabStopCounter, initializer?: Expression): string[] {
+        if (isObjectBindingPattern(pattern)) {
+            const oldTabstop = counter.tabstop;
+            const childCounter = { tabstop: oldTabstop };
+            const rootParam = getJSDocParamAnnotation(path, initializer, isJs, /*isObject*/ true, isSnippet, childCounter);
+            let childTags: string[] | undefined = [];
+            for (const element of pattern.elements) {
+                const elementTags = elementWorker(path, element, childCounter);
+                if (!elementTags) {
+                    childTags = undefined;
+                    break;
+                }
+                else {
+                    childTags = childTags.concat(elementTags);

```

Comment:
maybe `childTags.push(...elementTags)` ? (I really don't like `concat` but especially when it's a side-effect in the end anyway)